components/Weather/index.js

```diff
@@ -0,0 +1,76 @@
+import axios from "axios";
+import { useState, useEffect } from "react";
+import styled from "styled-components";
+import Icon from "../Icons";
+
+export default function Weather({ outivity }) {
+  const [weatherData, setWeatherData] = useState(null);
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  const fetchData = async () => {
+    try {
+      const response = await axios.get(
+        `https://api.openweathermap.org/data/2.5/weather?lat=${outivity.lat}&lon=${outivity.lng}&units=metric&appid=9e7a02e3cc30d92b7c137c9919b52688`
```

Comment:
Did you have to register for the weather API? I'm wondering if the appid is a secret information? Then you shoudn't use it inside the frontend. Maybe the number of fetches is restricted and someone else can now use your appid to fetch the weather api.

Reply:
We have transferred our key to env.local